### ISSUE TITLE
why3.withProvers : installation fails with "variable $src or $srcs should point to the source"

### ISSUE THREAD

**Issue by qaristote** [NixOS/nixpkgs#149402]
### Describe the bug
I'm using the unstable versions of NixOS and Nixpkgs. Running
```
nix-shell -p "why3.withProvers []"
```
fails with the trace
```
this derivation will be built:
  /nix/store/pychndqz867i7x06gppm7mkqnhd9ni26-why3-1.4.0-with-provers.drv
building '/nix/store/pychndqz867i7x06gppm7mkqnhd9ni26-why3-1.4.0-with-provers.drv'...
unpacking sources
variable $src or $srcs should point to the source
error: builder for '/nix/store/pychndqz867i7x06gppm7mkqnhd9ni26-why3-1.4.0-with-provers.drv' failed with exit code 1;
       last 2 log lines:
       > unpacking sources
       > variable $src or $srcs should point to the source
       For full logs, run 'nix log /nix/store/pychndqz867i7x06gppm7mkqnhd9ni26-why3-1.4.0-with-provers.drv'.
```

### Steps To Reproduce
1. Run
```
nix-shell -p "why3.withProvers []"
```

### Expected behavior
Enter a shell where `why3` is available.

### Additional context
This only fails when using `withProvers` : `nix-shell -p why3` works as expected.

### Notify maintainers
@thoughtpolice @vbgl

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.81, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(qaristote): `""`
 - channels(root): `"home-manager, nixos-22.05pre336298.bc5d68306b4, nixpkgs-22.05pre336340.cdaa4ce25b7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  why3
# a list of nixos modules affected by the problem
module:
```

**Comment by vbgl** [comment]
The error message comes from the `unpackPhase` added by #134361 (ping @Stunkymonkey). It can be fixed by adding `dontUnpack = true;` to the derivation.

There is an other issue though, probably brought during the update to 1.4.0: you may want to also try the following change:
~~~
-    why3 config --detect-provers -C $out/share/why3/why3.conf
+    why3 config detect -C $out/share/why3/why3.conf
~~~

**Comment by vbgl** [comment]
Fixed by #149459.
